.github/actions/pre-commit/action.yml

```diff
@@ -13,14 +13,22 @@ inputs:
     description: The python version of the local runner
     required: false
     default: "3.9"
+  auto-commit:
+    description: "enables auto-commit of eventual fixups (requires `permissions.contents: write` on the calling job)"
+    required: false
+    default: "false"
 runs:
   using: "composite"
   steps:
     - uses: actions/checkout@v3
+      with:
+        ref: ${{ github.head_ref }}
```

Comment:
I'm not 100% sure about this: I understand this is needed for the auto-commit behavior, but I wonder if that changes what pre-commit is checking on PRs, because it would match the source branch (maybe I'm confused, just want to make sure of this)

Reply:
should only affects workflow triggered by pull_request events (in all the other cases this var is empty - thus like not specifying any ref).

The difference is that pull_request by default run on the merge commit with the base branch (e.g. to see what happen once merged to main) and that is a detached head so you will lost any commit done on top of that.

adding head_ref make sure that git checkout the source branch and we get the expected behavior (additional commits appearing in the PR branch).

Reply:
I gathered that this would not be an issue for push events, I was just not sure we'd run pre-commit with additional commits on the PR branch, thanks for confirming :+1: